lib/Models/Newsletter.php

```diff
@@ -369,19 +369,41 @@ public function withOptions() {
   }
 
   public function wasScheduledForSubscriber($subscriberId) {
+    $query = SendingQueue::selectExpr('COUNT(*)', 'count');
+    $query = $this->getAllQueuesForSubscscriberQuery($query, $subscriberId);
     /** @var \stdClass */
-    $queue = SendingQueue::rawQuery(
-      "SELECT COUNT(*) as count
-      FROM `" . SendingQueue::$_table . "`
-      JOIN `" . ScheduledTask::$_table . "` ON " . SendingQueue::$_table . ".task_id = " . ScheduledTask::$_table . ".id
-      JOIN `" . ScheduledTaskSubscriber::$_table . "` ON " . ScheduledTask::$_table . ".id = " . ScheduledTaskSubscriber::$_table . ".task_id
-      WHERE " . ScheduledTaskSubscriber::$_table . ".subscriber_id = " . $subscriberId . "
-      AND " . SendingQueue::$_table . ".newsletter_id = " . $this->id
-    )->findOne();
+    $queue = $query->findOne();
 
     return ((int)$queue->count) > 0;
   }
 
+  private function getAllQueuesForSubscscriberQuery($orm, $subscriberId) {
```

Comment:
Here we add more usages of ORM we want to get rid of. Could you create a Doctrine repository instead? Automatic event class are not yet refactored to use DI container but you can use ContainerWrapper::getInstance()->ger() in the event class constructor as we do in other places that are not refactored yet.